prow/cmd/deck/static/spyglass/spyglass.ts

```diff
@@ -99,6 +113,39 @@ function toggleExpansion(bodyId: string, expanderId: string): void {
   }
 }
 
+// given a string containing ansi formatting directives, return a new one
+// with designated regions of text marked with the appropriate color directives,
+// and with all unknown directives stripped
+function ansiToHTML(orig: string) {
+  // Given a cmd (like "32" or "0;97"), some enclosed body text, and the original string,
+  // either return the body wrapped in an element to achieve the desired result, or the
+  // original string if nothing works.
+  function annotate(cmd: string, body: string, orig: string) {
+    var code = +(cmd.replace('0;', ''));
+    if (code === 0) // reset
+      return body;
+    else if (code === 1) // bold
+      return '<em>' + body + '</em>';
+    else if (30 <= code && code <= 37) // foreground color
+      return '<span class="ansi-' + (code - 30) + '">' + body + '</span>';
+    else if (90 <= code && code <= 97) // foreground color, bright
+      return '<span class="ansi-' + (code - 90 + 8) + '">' + body + '</span>';
+    return body;  // fallback: don't change anything
+  }
+  // Find commands, optionally followed by a bold command, with some content, then a reset command.
+  // Unpaired commands are *not* handled here, but they're very uncommon.
+  var filtered = orig.replace(/\033\[([0-9;]*)\w(\033\[1m)?([^\033]*?)\033\[0m/g, function(match: string, cmd: string, bold: string, body: string, offset: number, str: string) {
```

Comment:
:robot: `Prefer const where the value is never reassigned.`
:robot: `Prefer arrow functions.`

```suggestion
  const filtered = orig.replace(/\033\[([0-9;]*)\w(\033\[1m)?([^\033]*?)\033\[0m/g, (match: string, cmd: string, bold: string, body: string, offset: number, str: string) => {
```